src/main/java/software/amazon/smithy/gradle/tasks/SmithyCliTask.java

```diff
@@ -124,16 +207,23 @@ final void executeCliProcess(
         if (modelDiscoveryClasspath != null) {
             args.add("--discover-classpath");
             args.add(modelDiscoveryClasspath.getAsPath());
-        } else if (modelDiscovery) {
+        } else if (!disableModelDiscovery) {
             args.add("--discover");
         }
 
         args.addAll(customArguments);
 
-        java.util.Optional.ofNullable(getModels()).ifPresent(models -> {
+        if (!getModels().isEmpty()) {
             args.add("--");
-            models.forEach(file -> args.add(file.getAbsolutePath()));
-        });
+            getModels().forEach(file -> {
```

Comment:
Only add models that exist. I figure this is more like how the rest of Gradle works (you can just try to do stuff with files that don't exist). We can fail later on files that don't exist in the future if that's more desirable.